### ISSUE TITLE
Simple service registration seems not to work

### ISSUE THREAD

**Issue by stefan-kemp** [ivargrimstad/snoop#17]
I started the Snoop service in a Docker container (which is running in Vagrant because I'am working on a Windows machine) with the following command:

```
docker run -it -d -p 8081:8080 ivargrimstad/snoop-service
```

It starts successfully and is then accessible via the browser on: http://localhost:8081/snoop-service.

Next, I created a new project (source attached) and trying to start it in a Glassfish 4.1.1 instance. This fails, the output from IntelliJ:

```
C:\Tools\glassfish-4.1.1\glassfish\bin\asadmin.bat start-domain domain1
[2015-10-29 11:11:34,733] Artifact untitled5:war: Server is not connected. Deploy is not available.
Detected server admin port: 4848
Detected server http port: 8080
Attempting to start domain1.... Please look at the server log for more details.....
Connected to server
[2015-10-29 11:11:43,213] Artifact untitled5:war: Artifact is being deployed, please wait...
[2015-10-29 11:12:12,272] Artifact untitled5:war: Error during artifact deployment. See server log for details.
[2015-10-29 11:12:12,273] Artifact untitled5:war: java.io.IOException: com.sun.enterprise.admin.remote.RemoteFailureException: Error occurred during deployment: Exception while loading the app : javax.ejb.CreateException: Initialization failed for Singleton SnoopRegistrationClient. Please see server.log for more details.
```

Glassfish server log: https://www.dropbox.com/s/vokck0xl3sri3cl/server.log?dl=0
Source code: https://www.dropbox.com/s/j99azkh724ef9u9/source.zip?dl=0

**Comment by ivargrimstad** [comment]
Yes, I had this problem with GlassFish as well. It has something to do with the timer bean that is being registered. It works fine in WildFly 8,9,10 and Payara 4.1.
Same problem occurs with WebSphere Liberty as well as WebLogic 12.2.1.

_Will investigate further. May be a bug in GlassFish or some missing configuration..._

**Comment by ivargrimstad** [comment]
So, I found the issue. It is actually not related to timers as suggested in my previous comment. It is a result of different strategies for classloading in the application server implementations. See Mark's blog entry here: https://struberg.wordpress.com/category/cdi/

There is a workaround for Glassfish (and probably the others as well:
1. Copy the snoop jar (currently _snoop-1.3.0.jar_) to `$GLASSFISH_HOME/glassfish/domains/domain1/lib/ext`
2. Change the scope of the snoop dependency in your pom.xml to provided

```
<dependency>
   <groupId>eu.agilejava</groupId>
    <artifactId>snoop</artifactId>
    <version>${snoop.version}</version>
    <scope>provided</scope>
</dependency>
```

I will keep the issue open until we have managed a better way of solving this.
